eth2util/types.go

```diff
@@ -22,14 +22,20 @@ import (
 	"github.com/obolnetwork/charon/app/errors"
 )
 
+// SignedEpoch represents signature of corresponding epoch.
+type SignedEpoch struct {
+	Epoch     eth2p0.Epoch
+	Signature eth2p0.BLSSignature
+}
+
 // EpochHashRoot returns the ssz hash root of the epoch.
-func EpochHashRoot(epoch eth2p0.Epoch) ([32]byte, error) {
+func (s SignedEpoch) EpochHashRoot() ([32]byte, error) {
```

Comment:
can refactor this to align with eth2client pattern of getting the hash root of any type
<img width="980" alt="image" src="https://user-images.githubusercontent.com/29249923/186129054-00959dc0-6710-457c-99d3-72b299f85580.png">